engine/access/rpc/backend/backend_events.go

```diff
@@ -98,11 +103,33 @@ func (b *backendEvents) getBlockEventsFromExecutionNode(
 		BlockIds: convert.IdentifiersToMessages(blockIDs),
 	}
 
-	// call the execution node gRPC
-	resp, err := b.executionRPC.GetEventsForBlockIDs(ctx, &req)
+	var resp *execproto.GetEventsForBlockIDsResponse
+	var err error
+	if b.staticExecutionRPC != nil {
```

Comment:
I'm still not seeing how this is being set to nil. Is it suppose to be set to nil if `script-addr` is empty? because i don't think we do that yet do we?